src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -576,64 +593,97 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
                 8, 14, 13, 12,
                 -1, -1, -1, -1);
 
-            Vector128<sbyte> mask2F = Vector128.Create((sbyte)'/');
-            Vector128<sbyte> mergeConstant0 = Vector128.Create(0x01400140).AsSByte();
+            Vector128<byte> mask2F = Vector128.Create((byte)'/');
+            Vector128<byte> mergeConstant0 = Vector128.Create(0x01400140).AsByte();
             Vector128<short> mergeConstant1 = Vector128.Create(0x00011000).AsInt16();
-            Vector128<sbyte> zero = Vector128<sbyte>.Zero;
+            Vector128<byte> zero = Vector128<byte>.Zero;
+            Vector128<byte> one = Vector128.Create((byte)1);
+            Vector128<byte> f = Vector128.Create((byte)0xf);
 
             byte* src = srcBytes;
             byte* dest = destBytes;
 
             //while (remaining >= 24)
             do
             {
-                AssertRead<Vector128<sbyte>>(src, srcStart, sourceLength);
-                Vector128<sbyte> str = Sse2.LoadVector128(src).AsSByte();
+                Vector128<byte> str = Vector128.LoadUnsafe(ref *src);
 
                 // lookup
-                Vector128<sbyte> hiNibbles = Sse2.And(Sse2.ShiftRightLogical(str.AsInt32(), 4).AsSByte(), mask2F);
-                Vector128<sbyte> loNibbles = Sse2.And(str, mask2F);
-                Vector128<sbyte> hi = Ssse3.Shuffle(lutHi, hiNibbles);
-                Vector128<sbyte> lo = Ssse3.Shuffle(lutLo, loNibbles);
+                Vector128<byte> hiNibbles = Vector128.ShiftRightLogical(str.AsInt32(), 4).AsByte();
+                Vector128<byte> loNibbles = str;
+                if (Ssse3.IsSupported)
+                {
+                    hiNibbles = Vector128.BitwiseAnd(hiNibbles, mask2F);
+                    loNibbles = Vector128.BitwiseAnd(str, mask2F);
+                }
+                Vector128<byte> hi = SimdShuffle(lutHi, hiNibbles, f);
+                Vector128<byte> lo = SimdShuffle(lutLo, loNibbles, f);
 
                 // Check for invalid input: if any "and" values from lo and hi are not zero,
                 // fall back on bytewise code to do error checking and reporting:
-                if (Sse2.MoveMask(Sse2.CompareGreaterThan(Sse2.And(lo, hi), zero)) != 0)
-                    break;
+                if (Ssse3.IsSupported)
+                {
+                    if (Sse2.MoveMask(Vector128.GreaterThan(Vector128.BitwiseAnd(lo, hi), zero)) != 0)
+                        break;
+                }
+                else
+                {
+                    if (Vector128.BitwiseAnd(lo, hi).AsUInt64().ToScalar() != 0)
+                        break;
```

Comment:
It looks like this path is more efficient than Ssse3, isn't it?

Reply:
Here is the codegen difference on x64: https://www.diffchecker.com/WGoXdbb3 (MoveMask path is on the left) - so if the code is correct I am pretty sure you can delete the SSSE3 path, it's clearly worse